### PR TITLE
fix(extractors): align Java/Kotlin/CUDA contains parity across engines

### PR DESCRIPTION
**Pull request by carlos-alm**
## Summary

Three more per-language divergences from the v3.10.1-dev.80 dogfood report's residual `contains` gap, refs #1189.

- **Java native** — `handle_method_decl` did not skip methods inside an `interface_declaration`, so each interface method was emitted twice: once by `handle_interface_decl` (no children, matches WASM) and once again by the recursive walk's `handle_method_decl` (with parameter children). The duplicate inflated `contains` and `parameter_of` edge counts. Added the same `parent.parent == "interface_declaration"` guard the C# native extractor uses (introduced in #1194). 4 native-only contains edges in the fixture.

- **Kotlin WASM** — `collectKotlinMethods` and the object-body method loop in `handleKotlinObjectDecl` emitted class/object methods with no parameter children, while native correctly extracted them. WASM now calls `extractKotlinParameters` for each method and attaches the result as children. 8 native-only contains edges in the fixture.

- **Kotlin native** — `extract_kotlin_class_properties` looked for a `simple_identifier` directly inside `property_declaration`, but tree-sitter-kotlin nests the name as `property_declaration > variable_declaration > simple_identifier`. The lookup now drills through `variable_declaration` first (mirrors WASM's `collectKotlinProperties`). 1 WASM-only contains edge in the fixture.

- **CUDA both engines** — the tree-sitter-cuda grammar models a class body as a list of `field_declaration`s, so method declarations in `.cuh` headers were being emitted as `property` children whose name is the full method signature. Native stripped the pointer `*` while WASM kept it, producing 2 + 2 mismatched contains edges on the fixture. Both engines now detect when a field's declarator (after unwrapping pointer/reference/array/parenthesized wrappers) is a `function_declarator` and skip the entry — these are method declarations, not data fields. This also clears the unrelated "method-signature-as-property-name" bug for CUDA headers.

## Verification

- Cross-engine contains-edge diff on `tests/benchmarks/resolution/fixtures/{java,kotlin,cuda}/*` is now zero in both directions.
- `npx vitest run tests/engines/parity.test.ts` — 13 passed, 4 skipped (Java interface, Kotlin class methods, and CUDA header cases newly added as regression guards).
- `npx vitest run tests/parsers/ tests/engines/` — 567 passed.
- `npx vitest run tests/integration/ tests/graph/ tests/benchmarks/resolution/` — 975 passed.
- `cargo test --release --lib -- extractors::java extractors::kotlin extractors::cuda` — 78 passed.

## Test plan

- [x] Java interface method declarations get parameter children only via `handle_interface_decl` (no double emission).
- [x] Kotlin class/object methods carry parameter children in both engines.
- [x] Kotlin native `property_declaration` extraction sees the `variable_declaration` name.
- [x] CUDA `field_declaration`s whose declarator is a `function_declarator` are skipped in both engines.
- [x] All existing parity, parser, integration, and resolution tests pass.

Refs #1189 (Java, Kotlin, CUDA). Remaining: Objective-C, Ruby (single-edge case).

### CODE REVIEW

File: src/extractors/cuda.ts [hunk 1]
```diff
@@ -284,22 +284,43 @@ function extractCudaClassFields(classNode: TreeSitterNode): SubDeclaration[] {
...
+  'parenthesized_declarator',
+]);
+
+function isCudaMethodDeclarator(node: TreeSitterNode): boolean {
+  let current: TreeSitterNode | null = node;
+  while (current && CUDA_DECLARATOR_WRAPPERS.has(current.type)) {
+    current = current.childForFieldName('declarator');
+  }
+  return current?.type === 'function_declarator';
+}
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply src/extractors/cuda.ts:322]: Good catch — confirmed: `isCudaMethodDeclarator` cannot distinguish a method declaration from a function-pointer field (`void (*fp)(int)`) because both have a top-level `function_declarator` in tree-sitter-cuda. Parity is preserved (both engines skip), but a real function-pointer field would be silently dropped.

Out of scope for this PR — the stated goal is closing the residual Java/Kotlin/CUDA `contains` parity gap from the v3.10.1-dev.80 dogfood report, and a proper fix needs a dedicated function-pointer fixture + inner-declarator inspection. Tracked in #1204.

**Comment by carlos-alm** [comment]
@greptile
